compiler/rustc_typeck/src/check/method/suggest.rs

```diff
@@ -1167,22 +1168,53 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                 }),
                                 Applicability::MaybeIncorrect,
                             );
-                            suggested = true;
+                            return;
                         }
                         _ => {}
                     }
                 }
             }
 
-            if !suggested {
+            let (potential_candidates, explicitly_negative) = if param_type.is_some() {
+                // Negative bounds are not supported
```

Comment:
i don't feel like this comment as very helpful on its own, can you elaborate a bit here?

Reply:
I'll leave a TODO instead